server_src/utils/git.utils.js

```diff
@@ -0,0 +1,39 @@
+const simpleGit = require("simple-git");
+const git = simpleGit();
+
+function returnCurrentGitStatus() {
+  return git.status();
+}
+
+function isBranchUpToDate(status) {
+  return status.behind === 0;
+}
+
+function isBranchInfront(status) {
+  return status.ahead !== 0;
+}
+
+function doesBranchContainModifiedFiles(status) {
+  if (status.modified.length !== 0) {
+    return status.modified;
+  } else {
+    return false;
+  }
+}
+
+async function pullLatestRepository(force) {
+  if (force.forcePull) {
+    return git.pull();
+  } else {
+    await git.reset("hard");
```

Comment:
Why not provide forcePull directly?
Also, why does forcePull false trigger a reset? Not correct or unclear to me

Reply:
That was a mistake, I've corrected the boolean check. 

What do you mean provide it directly? put it in it's own function? Don't mind either way.

Reply:
`function someFunction(forcePull)` instead of `function someFunction(force)`

Reply:
Sorted